jep/223/README.adoc

```diff
@@ -269,6 +269,108 @@ The intent of this proposal is to improve overall security for Jenkins instances
+```
+public @CheckForNull Permission getRequiredPermission() {
+    return Jenkins.ADMINISTER;
+}
+```
+This change  only removes the links that are displayed in the "/manage" page for users that don't have the related permission.  Users with the `Jenkins.ADMINISTER` permission will not see any changes in the “/manage” page, and users with the `Jenkins.MANAGE` permission will only see links that have explicitly override the default implementation of `getRequiredPermission`.
+As this could break plugin compatibility, we did an analysis on the plugin usage of `getRequiredPermission` or `requiredPermission` in jelly/groovy.
+
+.Table Plugin usage analysis of ManagementLink.getRequiredPermission()
+|===
+|Plugin |Analyse |Reason
+
+|multi-slave-config-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|scriptler-plugin
+|No change
+|Manages its own permission in the ManagementLink
+
+|p4-plugin
+|No change
+|Based on scriptler-plugin
+
+|periodic-jobs-balancer-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|gerrit-trigger-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|lenient-shutdown-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|script-security-plugin
+|No change
+|getRequiredPermission() is overrided with `RUN_SCRIPT` in the ManagementLink
+
+|active-choices-plugin
+|No change
+|Based on scriptler-plugin
+
+|docker-plugin
+|Change
+|Does not override getRequiredPermission() in the ManagementLink and use it in jelly. The management page will be restricted to `ADMINISTER`.
+
+|libvirt-slave-plugin
+|Change
+|Does not override getRequiredPermission() in the ManagementLink and use it in jelly. The management page will be restricted to `ADMINISTER`.
+
+|build-with-parameters-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|metadata-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|flaky-test-handler-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|ssh-slaves-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|docker-traceability-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|rebuild-plugin
+|No related
+|The requiredPermission is used from an Action class, not a ManagementLink class.
+
+|global-build-stats-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|global-build-stats-plugin
+|No change
+|getRequiredPermission() is overrided with `ADMINISTER` in the ManagementLink
+
+|selenium-plugin
+|No change
+|getRequiredPermission() is overrided with a custom permission in the ManagementLink
+|===
+
```

Comment:
When looking for "equiredPermission" in the ecosystem, I got 191 occurences in 131 different files (java, groovy, jelly. It seems you did not cover all of them.

Second item on my list => build-metrics, two bugs, they expected to use the requiredPermission from ManagementLink in their views but the "it" was not the correct one and thus no impact.

Reply:
- dynamicparameter, two occurences:
The requiredPermission is asked on ScriptlerChoiceParameterDefinition.DescriptorImpl and ScriptlerStringParameterDefinition.DescriptorImpl but the classes and their respective hierarchies are missing the desired method/field.
=> That never worked since the initial commit and it's just a bug in their plugin.

Reply:
- support-core, lots of getRequiredPermissions, due to their Component approach, nothing impacting.

Reply:
- workflow-cps, same as support as it provides components to that plugin

Reply:
Thank you for your feedback @Wadeck I am going to add this last 4 plugins.